ggshield/scan/docker.py

```diff
@@ -41,6 +41,11 @@
 
 LAYER_TO_SCAN_PATTERN = re.compile(r"\b(copy|add)\b", re.IGNORECASE)
 
+IMAGE_NAME_PATTERN = (
+    r"^(?P<repository>[\w.\-_]+((?::\d+|)(?=/[a-z0-9._-]+/[a-z0-9._-]+))|)?(?:/|)"
+    r"(?P<image>[a-z0-9.\-_]+(?:/[a-z0-9.\-_]+|))(:(?P<tag>[\w.\-_]{1,127})?|)$"
+)
```

Comment:
This looks a bit complicated, do you think it would be enough to only check if the end of the name looks like a tag? I think a regular expression like this would work `:[a-zA-Z0-9_][-.a-zA-Z0-9_]{0,127}$`.

I made a few changes compared to your expression, based on the [docker tag doc](https://docs.docker.com/engine/reference/commandline/tag/#description):
- Allow 128 chars, but do not allow `.` and `-` on first char
- Use `a-zA-Z0-9` instead of `\w` because `\w` matches `_` and matches more than `a-zA-Z` unless `re.ASCII` is set (it matches `é` for example)

Also, `IMAGE_NAME_PATTERN` should be the result of `re.compile(<the_pattern>)` for consistency with the other `*PATTERN` constants.

Reply:
Done!